tests/checkers/unittest_imports.py

```diff
@@ -138,6 +138,42 @@ def test_preferred_module(capsys: CaptureFixture[str]) -> None:
         # assert there were no errors
         assert len(errors) == 0
 
+        # test preferred-modules case with submodules
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'preferred_module/unpreferred_submodule.py')}",
+                "-d all",
+                "-e preferred-module",
+                # prefer sys instead of os (for triggering test)
+                "--preferred-modules=os.path:pathlib",
```

Comment:
Could we also do a test with `os.something_that_is_not_path` and see if we don't still suggest it?

Reply:
Great idea, thank you! I'll add this into a further commit.

Reply:
Added in [9f42701](https://github.com/PyCQA/pylint/pull/8186/commits/9f4270151fde697ed8913818750435e9879cba3a)